blocks/search.py

```diff
@@ -104,35 +107,25 @@ def _compile_initial_state_computer(self):
         self.initial_state_computer = function(
             self.contexts, initial_states, on_unused_input='ignore')
 
-    def _compile_next_state_computer(self):
+    def _compile_states_and_logprobs_computer(self):
         next_states = [VariableFilter(bricks=[self.generator],
                                       name='^' + name + '$',
                                       roles=[OUTPUT])(self.inner_cg)[-1]
-                       for name in self.state_names]
-        next_outputs = VariableFilter(
-            application=self.generator.readout.emit, roles=[OUTPUT])(
-                self.inner_cg.variables)
-        self.next_state_computer = function(
-            self.contexts + self.input_states + next_outputs, next_states)
-
-    def _compile_logprobs_computer(self):
+                       for name in self.state_names_no_output]
         # This filtering should return identical variables
         # (in terms of computations) variables, and we do not care
         # which to use.
-        probs = VariableFilter(
+        logprobs = -tensor.log(VariableFilter(
```

Comment:
Okay, there are too many of you to simply ignore your opinion :) I will change it to `neglogprobs`.